azext_edge/edge/providers/check/base/pod.py

```diff
@@ -28,228 +28,186 @@ def decorate_pod_phase(phase: str) -> Tuple[str, str]:
-) -> None:
-    pod_health_status = "[green]Healthy[/green]"
-
-    if pod_eval_status == CheckTaskStatus.error.value:
-        pod_health_status = "[red]Unhealthy[/red]"
-    elif pod_eval_status == CheckTaskStatus.warning.value:
-        pod_health_status = "[yellow]Indeterminate[/yellow]"
-
-    pod_health_text = f"Pod {{[bright_blue]{pod_name}[/bright_blue]}} is {pod_health_status}"
+) -> Tuple[List[str], str]:
 
-    if detail_level != ResourceOutputDetailLevel.summary.value:
-        pod_health_text = f"\n{pod_health_text}"
+    def _decorate_pod_condition(condition: bool) -> Tuple[str, str]:
+        if condition:
+            return f"[green]{condition}[/green]", CheckTaskStatus.success.value
+        return f"[red]{condition}[/red]", CheckTaskStatus.error.value
 
-    check_manager.add_display(
+    target_service_pod = f"pod/{pod.metadata.name}"
+
+    pod_conditions = [
+        f"{target_service_pod}.status.phase",
+        f"{target_service_pod}.status.conditions",
+    ]
+
+    if check_manager.targets.get(target, {}).get(namespace, {}).get("conditions", None):
+        check_manager.add_target_conditions(target_name=target, namespace=namespace, conditions=pod_conditions)
+    else:
+        check_manager.set_target_conditions(target_name=target, namespace=namespace, conditions=pod_conditions)
+
+    pod_dict = pod.to_dict()
+    pod_name = pod_dict["metadata"]["name"]
+    pod_phase = pod_dict.get("status", {}).get("phase")
+    pod_conditions: list = pod_dict.get("status", {}).get("conditions", [])
+    pod_phase_deco, status = decorate_pod_phase(pod_phase)
+
+    pod_eval_value = {}
+    pod_eval_status = status
+    pod_eval_value["status.phase"] = pod_phase
+
+    conditions_readiness = True
+    conditions_display_list: List[Tuple[str, str]] = []
+    unknown_conditions_display_list: List[Tuple[str, str]] = []
+
+    # When pod in obnormal state, sometimes the conditions are not available
+    if pod_conditions:
+        for condition in pod_conditions:
+            type = condition["type"]
+            condition_type = POD_CONDITION_TEXT_MAP.get(type)
+
+            if condition_type:
+                condition_status = condition.get("status") == "True"
+                conditions_readiness = conditions_readiness and condition_status
+                pod_condition_deco, status = _decorate_pod_condition(condition=condition_status)
+                pod_eval_status = status if status != CheckTaskStatus.success.value else pod_eval_status
+            else:
+                condition_type = type
+                condition_status = condition.get("status")
+
+            formatted_reason = ""
+            condition_reason = condition.get("reason", "")
+
+            if condition_reason:
+                formatted_reason = f"[red]Reason: {condition_reason}[/red]"
+
+            known_condition_values = [value.replace(" ", "").lower() for value in POD_CONDITION_TEXT_MAP.values()]
```

Comment:
This should probably be stored outside of the loop, no need to regenerate this list every time